app/uk/gov/hmrc/economiccrimelevyreturns/controllers/CheckYourAnswersController.scala

```diff
@@ -50,86 +48,99 @@ class CheckYourAnswersController @Inject() (
-
-  private def contactDetails()(implicit request: ReturnDataRequest[_]): SummaryList = SummaryListViewModel(
-    rows = Seq(
-      ContactNameSummary.row(),
-      ContactRoleSummary.row(),
-      ContactEmailSummary.row(),
-      ContactNumberSummary.row()
-    ).flatten
-  ).withCssClass("govuk-!-margin-bottom-9")
-
-  private def amendReasonDetails()(implicit request: ReturnDataRequest[_]): SummaryList = SummaryListViewModel(
-    rows = Seq(
-      AmendReasonSummary.row()
-    ).flatten
-  ).withCssClass("govuk-!-margin-bottom-9")
-
   def onPageLoad: Action[AnyContent] = (authorise andThen getReturnData).async { implicit request =>
-    val isAmendment = request.eclReturn.returnType.contains(AmendReturn)
     (for {
       errors <- returnsService.getReturnValidationErrors(request.internalId)(hc).asResponseError
-    } yield errors).fold(
-      error => routeError(error),
-      {
-        case Some(_) => Redirect(routes.NotableErrorController.answersAreInvalid())
-        case None    => Ok(view(amendReasonDetails(), eclDetails(), contactDetails(), isAmendment, request.startAmendUrl))
-      }
-    )
+    } yield errors)
+      .fold(
+        error => Future.successful(routeError(error)),
+        {
+          case Some(_) => Future.successful(Redirect(routes.NotableErrorController.answersAreInvalid()))
+          case None    =>
+            viewHtmlOrError().fold(
+              error => routeError(error),
+              view => Ok(view)
+            )
+        }
+      )
+      .flatten
   }
 
   def onSubmit: Action[AnyContent] = (authorise andThen getReturnData).async { implicit request =>
     implicit val hc: HeaderCarrier = CorrelationIdHelper.getOrCreateCorrelationId(request)
 
-    val isAmendment = request.eclReturn.returnType.contains(AmendReturn)
-    val htmlView    = view(amendReasonDetails(), eclDetails(), contactDetails(), isAmendment)
-
-    val base64EncodedHtmlView: String = base64EncodeHtmlView(htmlView.body)
-
-    val base64EncodedDmsSubmissionHtml = request.eclReturn.returnType.flatMap {
-      case AmendReturn     => Some(createAndEncodeHtmlForPdf())
-      case FirstTimeReturn => None
-    }
+    (for {
+      viewHtml <- viewHtmlOrError()
+    } yield viewHtml)
+      .fold(
+        error => Future.successful(routeError(error)),
+        viewHtml =>
+          (for {
+            pdfViewHtml <- pdfViewHtmlOrError()
+          } yield pdfViewHtml)
+            .fold(
+              error => Future.successful(routeError(error)),
+              pdfViewHtml => {
 
-    val updatedReturn = request.eclReturn.copy(
-      base64EncodedNrsSubmissionHtml = Some(base64EncodedHtmlView),
-      base64EncodedDmsSubmissionHtml = base64EncodedDmsSubmissionHtml
-    )
+                val updatedReturn = request.eclReturn.copy(
+                  base64EncodedNrsSubmissionHtml = Some(base64EncodeHtmlView(viewHtml.body)),
+                  base64EncodedDmsSubmissionHtml = pdfViewHtml match {
```

Comment:
you could just map the option: 

pdfViewHtml.map(html =>base64EncodeHtmlView(html.body) )

Reply:
Done